src/NuGet.Core/NuGet.Protocol/Resources/PackageSearchResourceV3.cs

```diff
@@ -142,7 +142,7 @@ private async Task<T> SearchPage<T>(
                 {
                     var types = string.Join("&",
                         filters.PackageTypes.Select(
-                            s => "packageTypeFilter=" + s));
+                            s => "packageType=" + s));
                     queryString += "&" + types;
```

Comment:
Only a single package type filter is supported. The package type parameter is plumbed through in our search service as a single string:
https://github.com/NuGet/NuGetGallery/blob/dc7abf2bd145d91596b032c19e1c9abe8b276956/src/NuGet.Services.AzureSearch/SearchService/SearchParametersBuilder.cs#L169-L172

PRs on the docs are welcome if they are unclear!

Reply:
Regarding multiple package types on the package, the package type parameter matches the package if ANY of the package types match, per:
> The `packageType` parameter is used to further filter the search results to only packages **that have at least one package type matching the package type name**.

Emphasis mine.

Reply:
@joelverhagen @zivkan Thank you both for the feedback. In that case, it seems that the current API surface of `SearchFilter` might be inconsistent with current `SearchQueryService`, as it allows for multiple package types to be specified.

How do you prefer to address this? Shall `PackageSearchResourceV3` and `LocalPackageSearchResource` throw if more than one package type is provided? Or shall we deprecate the `PackageTypes` property and introduce a new `PackageType` property with a single value?

@joelverhagen Is there any expectation that multiple package type filters will ever be supported? That might inform whether the deprecation route might be preferred.

Reply:
@zivkan One last API design question which just occurred to me: what should the behavior of `LocalPackageSearchResource` be if the package type `Dependency` is specified and the local package does not specify package types, e.g. for older packages without the `packageTypes` metadata?

In [the docs](https://learn.microsoft.com/en-us/nuget/create-packages/set-package-type?tabs=dotnet#known-package-types) it is stated that _"Packages not marked with a type, including all packages created with earlier versions of NuGet, default to the `Dependency` type."_

This would suggest that we should accept matches for all packages without `packageTypes` metadata if the specified package type filter is `Dependency`.

Reply:
There is no expectation that NuGet.org should support multiple package type filters. If this is needed, we would need to design, implement, and document a protocol enhancement. I am not aware of any other package source that implements multiple package type filters.

IMHO, `PackageTypes` should be deprecated and `PackageType` should be introduced, to match the current state of the protocol. But I do not work on the client side much so I may be missing something.

Reply:
I don't know of any plans to add multiple package type support. It's not even clear to what existing scenario could benefit from such a filter.

Reply:
> Personally I would prefer not touching the current surface API

To be fair, this PR is breaking server APIs, even if it's not breaking the assembly's API. If there are any servers (that are not nuget.org) that use the current `packageTypeFilter`, this PR will break clients using those servers. [There's evidence](https://github.com/NuGet/NuGetGallery/issues/5999) that (in 2018) neither nuget.org nor myget implemented `packageTypeFilter`, but there's no way to know if some other server uses it. On the other hand, the V2 protocol isn't documented, and `packageTypeFilter` was never part of [the official V3 protocol](https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource).

I'm trying to rush reviewing a bunch of old PRs that I haven't reviewed in a long time, so I haven't yet formed an opinion about breaking APIs (about changing NuGet.Protocol to remove the plural `PackageTypes` and have a singular `PackageType`). There's a case to be made that NuGet.Client should implement the official protocol spec.